### PR TITLE
EQL: Add ? support (match single char) to wildcard functionality in  `:` operator

### LINKED ISSUE #65536: EQL: Add ? support (match single char) to wildcard functionality in  `:` operator
Wildcard pattern supported by `:` currently checks only for `*` but `?` should be supported as well.

### PR DESCRIPTION
**Pull request by costin**
Introduce support for ? character to wildcard pattern to : operator and
by transition, to wildcard function.

Fix #65536

### CODE REVIEW

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/optimizer/OptimizerTests.java [hunk 1]
```diff
@@ -152,6 +152,27 @@ public void testEqualsWildcardOnRight() {
         }
     }
 
+    public void testEqualsWildcardQuestionmarkOnRight() {
+        List<String> tests = Arrays.asList(
```
Comment (astefan) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/optimizer/OptimizerTests.java:156]: Why not defining a list when it's actually needed? Or add one more query. It's a bit too much to use a list for just one query imo. I remember there is one more test like this for `*` (using a list for one query only).

**Review by astefan — APPROVED** [review]
LGTM. Left one comments.

**Review by matriv — COMMENTED** [review]
LGTM, Could we have some integ test with `wildcard` as well, for completness?

File: x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml [hunk 2]
```diff
@@ -300,3 +305,24 @@ query = '''
...
+name = "wildcardFunctionWildcardPattern"
+query = '''
+file where wildcard(file_name, "winini*.exe", "lsass.*") and opcode == 2
+'''
+expected_event_ids  = [65, 86]
+
+[[queries]]
+name = "wildcardFunctionQuestionMarkPattern"
+query = '''
+file where file_name : ("winini?.exe", "lsass.e?e") and opcode == 2
```
Comment (matriv) [inline x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml:326]: This should be `wildcard` func instead, no?

**Review by matriv — APPROVED** [review]
LGTM, sorry one more comment on the new test.

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/StringUtilsTests.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+public class StringUtilsTests extends ESTestCase {
+
+    public void testLikePatternNoPattern() throws Exception {
+        String string = "abc123";
+        assertEquals(string, toLikePattern(string).asString());
+    }
+
+    public void testLikePatternLikeChars() throws Exception {
+        String string = "a%bc%%12_3__";
+        String escape = Character.toString(1);
```
Comment (bpintea) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/StringUtilsTests.java:26]: Not strictly related to the change, but was wondering the value choice for the escape character (in `StringUtils#toLikePattern()`). Would `\0` have collided with anything?

**Review by bpintea — APPROVED** [review]
LGTM

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/StringUtilsTests.java (hunk 3 above)
Comment (costin) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/StringUtilsTests.java:26]: 0 ignores escaping, that is no character acts as an escape.

**Pull request merged**